color/src/gradient.rs

```diff
@@ -26,26 +26,84 @@ pub struct GradientIter<CS: ColorSpace> {
 
 /// Generate a piecewise linear approximation to a gradient ramp.
 ///
-/// A major feature of CSS Color 4 is to specify gradients in any
-/// interpolation color space, which may be quite a bit better than
-/// simple linear interpolation in sRGB (for example).
-///
-/// One strategy for implementing these gradients is to interpolate
-/// in the appropriate (premultiplied) space, then map each resulting
-/// color to the space used for compositing. That can be expensive.
-/// An alternative strategy is to precompute a piecewise linear ramp
-/// that closely approximates the desired ramp, then render that
-/// using high performance techniques. This method computes such an
-/// approximation.
-///
-/// The given `tolerance` value specifies the maximum error in the
-/// approximation, in deltaEOK units. A reasonable value is 0.01,
-/// which in testing is nearly indistinguishable from the exact
-/// ramp. The number of stops scales roughly as the inverse square
-/// root of the tolerance.
-///
-/// The error is measured at the midpoint of each segment, which in
-/// some cases may underestimate the error.
+/// The target gradient ramp is the linear interpolation from `color0` to `color1` in the target
+/// color space specified by `interp_cs`. For efficiency, this returns an
+/// [iterator over color stops](GradientIter) in the `CS` color space, such that the gradient ramp
+/// created by linearly interpolating between those stops in the `CS` color space is equal within
+/// the specified `tolerance` to the target gradient ramp.
+///
+/// When the target interpolation color space is cylindrical, the hue can be interpolated in
+/// multiple ways. The [`direction`](`HueDirection`) parameter controls the way in which the hue is
+/// interpolated.
+///
+/// The given `tolerance` value specifies the maximum error in the approximation, in deltaEOK
```

Comment:
I think we should link to something to explain terms like "deltaEOK" ... perhaps https://www.w3.org/TR/css-color-4/#color-difference-OK ?

Reply:
I agree. I've added an explanation and linked to that section in the spec. Perhaps some of this stuff should be expanded on elsewhere in the future: for now we just have `gradient` where this is used, but it's also relevant for tone mapping.